### PR TITLE
Add basic UEFI code to load memory map

### PR DESCRIPTION
**Pull request by colescott**

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -6,7 +6,15 @@ stdenv.mkDerivation rec {
     name = name;
     paths = buildInputs;
   };
-  buildInputs = [
+  buildInputs = with pkgs; [
     (import ./tools/gcc.nix)
+    (import ./tools/binutils.nix)
+    qemu OVMF
```
Comment (eliaslfox) [inline default.nix]: this needs a newline

File: uefi/Makefile [hunk 2]
```diff
@@ -0,0 +1,50 @@
+BUILD_DIR=build
+SRC_DIR=src
+TARGET=x86_64
+
+CC=gcc
```
Comment (eliaslfox) [inline uefi/Makefile]: Is this compatible with non gnu systems?

File: .astylerc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+style=mozilla
```
Comment (eliaslfox) [inline .astylerc:1]: I don't like mozilla

**Review by eliaslfox — CHANGES_REQUESTED** [review]

**Review by eliaslfox — DISMISSED** [review]

**Review by eliaslfox — CHANGES_REQUESTED** [review]
-Werror

**Review by eliaslfox — DISMISSED** [review]

**Review by eliaslfox — APPROVED** [review]
Although I have strong reservations on the indentation of just about all of the c code in this pr. I think that the make files alone are dank enough to justify merging.

**Review by rizend — APPROVED** [review]

**Pull request merged**